pipert2/core/base/synchronize_routines/synchronizer_node.py

```diff
@@ -0,0 +1,125 @@
+from typing import List
+import multiprocessing as mp
+from pipert2.utils.consts import UPDATE_FPS_NAME
+
+
+class SynchronizerNode:
+    """ The synchronize node steps -
+    For synchronize the fps through the graph, we need to synchronize each sub logic with each other.
+
+    - The first step is updating the real fps of the current node.
+    - The second step is calculating the minimum fps between {all the sub logics, take their max value}.
+        When taking the max value of the sub logics, we handle the max fps that necessary.
+    - The third step is calculating the minimum fps between {all the node owners, take their max value}.
+        When taking the max value of the node owner, we handle the max fps that necessary from the owners.
+    - The forth step is notifying the fps.
+    - The fifth step is reset the flags. Because some node can be shared between two owners, and it already notified or
+        updated with the realtime fps, then raise a flag for not notifying or updating twice.
```

Comment:
```suggestion
    - The first step is to update the real fps of the current node.
    - The second step is to calculate the minimum fps between {all the sub logics, take their max value}.
        When taking the max value of a sub logic, we handle the max fps that is necessary.
    - The third step is to calculate the minimum fps between {all the node owners, take their max value}.
        When taking the max value of a node owner, we handle the max fps that is necessary.
    - The forth step is to notify the fps.
    - The fifth step is to reset the flags. Because some nodes can be shared between two owners, to avoid repeated functionality, we need to set the flags to notify that the node was updated.
```